taipy/core/data/_data_manager.py

```diff
@@ -29,21 +29,6 @@ class _DataManager(_Manager[DataNode]):
     _repository = _DataRepository(__DATA_NODE_CLASS_MAP)
     _ENTITY_NAME = DataNode.__name__
 
-    @classmethod
-    def _get_or_create(
-        cls,
-        data_node_config: DataNodeConfig,
-        scenario_id: Optional[ScenarioId] = None,
-        pipeline_id: Optional[PipelineId] = None,
-    ) -> DataNode:
-        scope = data_node_config.scope
-        parent_id = pipeline_id if scope == Scope.PIPELINE else scenario_id if scope == Scope.SCENARIO else None
-
-        if dn_from_parent := cls._repository._get_by_config_and_parent_id(data_node_config.id, parent_id):
-            return dn_from_parent
-
-        return cls._create_and_set(data_node_config, parent_id)
-
     @classmethod
     def _bulk_get_or_create(
         cls,
```

Comment:
Would it make more sense to pass `data_node_configs` as a `Set[DataNodeConfig]` instead of an Iterable?
The return type could be a `Dict[DataNodeConifg, DataNode]`. What do you think?